### ISSUE TITLE
Adding more features to the auto encoder

### ISSUE THREAD

**Issue by mulangonando** [minghao-wu/CRF-AE#4]
Hi,

I see there are only 3 categories of features represented for the autoencoder.
Is it possible to add more categories of features and how would I possibly use both categorical and other non categorical ones?

Thanks

**Comment by mulangonando** [comment]
Also if I wanted to access only the autoencoder outputs, which line do I really need to change?

**Comment by minghao-wu** [comment]
You are very welcomed to try any kinds of features, but I think you should be aware of three things.
1. Not all the features can help the model. Some does, some doesn't. It depends on your primary task.
2. You have to tune the lambda for each loss carefully. One of my intern once told me this idea didn't work on a certain task and finally it turned out that's due to bad choice of lambdas.
3. Choose proper loss functions, such as cross entropy for categorical features, MSE for continuous features.

You can get autoencoder outputs from [here](https://github.com/minghao-wu/CRF-AE/blob/master/model.py#L189) with some necessary modifications.

Good Luck and Happy Training.

**Comment by minghao-wu** [comment]
I will close this issue if you are happy with previous reply.

**Comment by nofelmahmood** [comment]
@minghao-wu I have modified the dimension of hidden layers to accommodate our POS and Shape features but I still get an error "Input size -1 must be equal to input size" on this line
```
// (Line 120 in model.py in method _get_lstm_features(self, sentence, chars2, caps, chars2_length, d, feature, gazetteer))
lstm_out, _ = self.lstm(embeds)
```

Am I missing something?

**Comment by minghao-wu** [comment]
I guess the dimensionality for the last dimension is incorrect. The last dimension of `embeds` should match the `input_size` of LSTM.

You should modify [this line](https://github.com/minghao-wu/CRF-AE/blob/480aa78eb2f761c074f2a83c504b18091dc0d2bd/model.py#L83) accordingly to make the model fit your own features.
